packages/domain/source/shared/todos/rules/unique-todo.js

```diff
@@ -0,0 +1,29 @@
+Space.domain.Rule.extend('Todos.UniqueTodoRule', {
+
+  mixin: [
+    Space.messaging.CommandSending
+  ],
+
+  enforceMap() {
+    return [{
+      'Todos.CreateTodo': this._onCreateTodo
+    }]
+  },
+
+  _onCreateTodo(command) {
+    console.log(
+      ['rule', 'before'], 'Todos.UniqueTodoRule', command.toString(),
+      ['S', 'C']
+    )
+
+    this.send(
+      new Todos.ValidateTodo({title: command.title}), function(err, result) {
+        if (result === false) {
```

Comment:
> Yes I'm calling them queries in my code. I think it would be nice to have syntax sugar like Space.messaging.Query that is defined as Commands - with fields for validation.

For now maybe just use a vanilla (non-command) messaging.Api endpoint. There's less need for a structured argument, and it's a familiar pattern without the need to introduce more classes